Formula/clickhouse-cpp.rb

```diff
@@ -17,18 +17,30 @@ class ClickhouseCpp < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "59b4a9754fd3934831f20c2efe2398c4a56b4e7fb0de43aff7a50642e80b4bf6"
   end
 
-  depends_on "cmake" => [:build, :test]
+  depends_on "cmake" => :build
   depends_on "abseil"
+  depends_on "cityhash"
+  depends_on "lz4"
   depends_on "openssl@3"
 
   fails_with gcc: "5"
   fails_with gcc: "6"
 
   def install
-    system "cmake", "-S", ".", "-B", "build",
-                    "-DWITH_OPENSSL=ON",
-                    "-DOPENSSL_ROOT_DIR=#{Formula["openssl@3"].opt_prefix}",
-                    *std_cmake_args
+    # `cityhash` does not provide a pkg-config or CMake config file.
+    # Help CMake find it.
+    inreplace "CMakeLists.txt", "FIND_PACKAGE(cityhash REQUIRED)",
+                                "FIND_LIBRARY(CITYHASH NAMES cityhash REQUIRED)"
+    inreplace "clickhouse/CMakeLists.txt", "cityhash::cityhash", "cityhash"
```

Comment:
Can we upstream this? Or has upstream forked cityhash to provide a CMake config file?

Reply:
Upstream has written a simple CMake config file for `cityhash`: https://github.com/ClickHouse/clickhouse-cpp/blob/master/contrib/cityhash/cityhash/CMakeLists.txt.

But since `cityhash` does not provide any CMake modules, the setup with `WITH_SYSTEM_CITYHASH=ON` is basically broken. I will upstream a proper fix.

Reply:
> Upstream has written a simple CMake config file for `cityhash`: https://github.com/ClickHouse/clickhouse-cpp/blob/master/contrib/cityhash/cityhash/CMakeLists.txt.

That's what I thought. This project likes to just fork all their dependencies.